src/cfnlint/decode/__init__.py

```diff
@@ -16,6 +16,52 @@
 LOGGER = logging.getLogger(__name__)
 
 
+def decode_str(s):
```

Comment:
Seems like repetitive logic which is unfortunate but looking at the structure of the current code I understand why you did this.

Reply:
What about doing something like this? 

https://github.com/kddejong/cfn-python-lint/blob/LaurenceWarne-new-api/src/cfnlint/decode/__init__.py#L19-L104

Reply:
Yeah lgtm, I've pushed ^ with a few minor tweaks, I think I was overthinking re logging since it's unconditionally disabled in `lint(s)` anyway.